src/ac_power_flow_jacobian.jl

```diff
@@ -1,228 +1,475 @@
-struct PolarPowerFlowJacobian{F,
-    T <: SparseArrays.SparseMatrixCSC{Float64, Int32},
-}
-    Jf::F
-    Jv::T
+"""
+    struct ACPowerFlowJacobian
+
+A struct that represents the Jacobian matrix for AC power flow calculations.
+
+This struct uses the functor pattern, meaning instances of `ACPowerFlowJacobian` store the data (Jacobian matrix) internally
+and can be called as a function at the same time. Calling the instance as a function updates the stored Jacobian matrix.
+
+
+# Fields
+- `data::ACPowerFlowData`: The grid model data used for power flow calculations.
+- `Jf!::Function`: A function that calculates the Jacobian matrix inplace.
+- `Jv::SparseArrays.SparseMatrixCSC{Float64, Int32}`: The Jacobian matrix, which is updated by the function `Jf!`.
+"""
+struct ACPowerFlowJacobian
+    data::ACPowerFlowData
+    Jf!::Function   # This is the function that calculates the Jacobian matrix and updates Jv inplace
+    Jv::SparseArrays.SparseMatrixCSC{Float64, Int32}  # This is the Jacobian matrix, that is updated by the function Jf
 end
 
-function (Jacobianf::PolarPowerFlowJacobian)(
-    J::SparseArrays.SparseMatrixCSC{Float64, Int32},
-    x::Vector{Float64},
-)
-    Jacobianf.Jf(Jacobianf.Jv, x)
-    copyto!(J, Jacobianf.Jv)
+"""
+    (J::ACPowerFlowJacobian)(time_step::Int64)
+
+Update the Jacobian matrix `Jv` using the function `Jf!` and the provided data and time step.
+
+Defining this method allows an instance of `ACPowerFlowJacobian` to be called as a function, following the functor pattern.
+
+# Arguments
+- `time_step::Int64`: The time step for the calculations.
+
+# Example
+```julia
+J = ACPowerFlowJacobian(data, time_step)
+J(time_step)  # Updates the Jacobian matrix Jv
+```
+"""
+function (J::ACPowerFlowJacobian)(time_step::Int64)
```

Comment:
we need to open an issue to change this interface where we pass the time_step here and change this interface.